src/runtime/string.cpp

```diff
@@ -251,7 +251,26 @@ static PRIMFN(prim_write) {
   REQUIRE(mpz_cmp_si(mode, 0x1ff) <= 0);
   long mask = mpz_get_si(mode);
 
-  deep_unlink(AT_FDCWD, path->c_str());
+  // We want `write` to overwrite existing files so that
+  // each time the build runs it isn't blocked by previous files.
+  // However we want it to fail if it attempts to delete a directory.
+  // We need to give the user a good error message in that case.
+  if (unlink(path->c_str()) < 0 && errno != ENOENT) {
+    if (errno == EISDIR) {
+      std::string error = path->c_str();
+      error +=
+          " is a directory and cannot be overwritten. If this is intentional please manually "
+          "delete this directory";
+      size_t len = std::min(error.size(), max_error);
+      String *out = String::claim(runtime.heap, error.c_str(), len);
```

Comment:
This is running into `max_error` problems, and really doesn't look good when it does: `Fail (Error "src is a directory and cannot be overwritten. If this is intentional please manually delete this direct" Nil)`

Reply:
oh crap, thanks for finding and debugging that! We should increase max error.

Reply:
I think there's always going to be some problem if we simply bump the value.  Using an unrealistically low `max_error` to stand in for a longer path than I want to type: `Fail (Error "very/long/path/to/some/fi" Nil)`.  We need to trim the filepath (with some indication that it's been trimmed) without risking the error message itself.

Reply:
I probably won't be prioritizing that level of quality fix but I'd love to see that sort of thing go in. We can add in the the size of the path instead I think. Paths on Linux are not allowed to be any longer than 4096 so we can set it to something like 100 + std::max(path.size(), 4096).